dot/digest/digest.go

```diff
@@ -201,17 +201,23 @@ func (h *Handler) handleBabeConsensusDigest(digest scale.VaryingDataType, header
 	case types.BABEOnDisabled:
 		return nil
 
-	case types.NextConfigData:
-		currEpoch, err := h.epochState.GetEpochForBlock(header)
+	case types.VersionedNextConfigData:
+		nextVersionedConfigData := digestValue.(types.VersionedNextConfigData)
+		nextConfigDataVersion, err := nextVersionedConfigData.Value()
 		if err != nil {
-			return fmt.Errorf("cannot get epoch for block %d (%s): %w",
-				header.Number, headerHash, err)
+			return fmt.Errorf("getting digest value: %w", err)
 		}
 
-		nextEpoch := currEpoch + 1
-		h.epochState.StoreBABENextConfigData(nextEpoch, headerHash, val)
-		h.logger.Debugf("stored BABENextConfigData data: %v for hash: %s to epoch: %d", digest, headerHash, nextEpoch)
-		return nil
+		if nextConfigData, ok := nextConfigDataVersion.(types.NextConfigDataV1); ok {
```

Comment:
would be nice to use a switch case with type assertion just as we are using + including an error if the version is not supported yet
```suggestion
		switch nextConfigData := nextConfigDataVersion.(type) {
		case types.NextConfigDataV1:
		   ...
		default:
		   return errors.New("next config data version not supported: %T", nextConfigDataVersion)
		}
```

Reply:
I did it but the `DeepSource: Go` complained. Seems to be a bad practice to use a switch with only one case 🤔

Reply:
Seems to be passing with the default case, thanks!